src/__swaps__/screens/Swap/hooks/useEstimatedGasFee.ts

```diff
@@ -35,20 +43,68 @@ export function useEstimatedGasFee({
   return useMemo(() => {
     if (!gasLimit || !gasSettings || !nativeNetworkAsset?.price) return;
 
-    const amount = gasSettings.isEIP1559 ? add(gasSettings.maxBaseFee, gasSettings.maxPriorityFee) : gasSettings.gasPrice;
-
-    const totalWei = multiply(gasLimit, amount);
+    const fee = calculateGasFee(gasSettings, gasLimit);
     const networkAssetPrice = nativeNetworkAsset.price.value?.toString();
 
-    if (!networkAssetPrice) return `${formatNumber(weiToGwei(totalWei))} Gwei`;
+    if (!networkAssetPrice) return `${formatNumber(weiToGwei(fee))} Gwei`;
 
-    const gasAmount = formatUnits(safeBigInt(totalWei), nativeNetworkAsset.decimals).toString();
-    const feeInUserCurrency = multiply(networkAssetPrice, gasAmount);
+    const feeFormatted = formatUnits(safeBigInt(fee), nativeNetworkAsset.decimals).toString();
+    const feeInUserCurrency = multiply(networkAssetPrice, feeFormatted);
 
     return formatCurrency(feeInUserCurrency);
   }, [gasLimit, gasSettings, nativeNetworkAsset]);
 }
 
+const getUserHasEnoughFundsForTx = (quote: Quote, gasFee: string) => {
```

Comment:
this will check for when the native asset is the asset you're selling - what behavior are you expecting for when a user is selling a non-native asset but types in a value much higher than their balance? we should still prevent them from proceeding to the review sheet

Reply:
![simulator_screenshot_EFAD520F-CB6A-4D04-AD93-92128BD2B876](https://github.com/rainbow-me/rainbow/assets/1285228/79cbc524-212b-4cda-9034-f5f3a56ce1f6)

Reply:
example: selling a non-native token (e.g, DAI) and bumping up the amount still lets me see the Review button instead of Insufficient Funds